### ISSUE TITLE
Export and import currencies

### ISSUE THREAD

**Issue by zorun** [spiral-project/ihatemoney#806]
Currently, export and import don't handle bill currencies.

We should add it, but there are several corner cases. In addition, ihatemoney 5 should accept to import a dump from ihatemoney 4 (without any currency information)

I'm working on it.

**Comment by zorun** [comment]
So, the plan is to add the currency in the bill export and the "settle plan" export with the list of transactions.

Here goes an example for the list of bills in CSV:
```
what,amount,currency,date,payer_name,payer_weight,owers
AMAP,50.0,EUR,2021-07-18,Amina,1.0,"Amina, Alice, Georg"
Beer !,20.0,EUR,2021-07-18,Alice,1.0,"Amina, Alice"
Food shopping,200.0,EUR,2021-07-18,Georg,1.0,"Amina, Georg, Alice"
```

And for a project without currency, it would look like this:
```
what,amount,currency,date,payer_name,payer_weight,owers
AMAP,50.0,XXX,2021-07-18,Amina,1.0,"Amina, Alice, Georg"
Beer !,20.0,XXX,2021-07-18,Alice,1.0,"Amina, Alice"
Food shopping,200.0,XXX,2021-07-18,Georg,1.0,"Amina, Georg, Alice"
```

I find it a bit ugly to expose this `XXX` thing, so I tried to represent "no currency" with an empty string when exporting while keeping it `XXX` internally. But it's a mess, we have to convert from `""` to `XXX` in many places and it's very easy to get it wrong in the code.

Any thoughts? I would really prefer to have a single string representation of "no currency". Could we switch to `""` internally?

**Comment by Glandos** [comment]
"XXX" is the [official ISO code for "No currency"](https://en.wikipedia.org/wiki/ISO_4217). That's why it is used, and in my opinion, should be used for export.
But the whole column can be stripped if the **default** currency is "XXX". And on import, if there is no currency column, then the project is using "XXX" as the default.

**Comment by zorun** [comment]
Ah right, I forgot that it's an official code, so I guess it's fine to expose it.

For exporting, I would prefer if the format is the same for all projects, it's much less confusing.

**Comment by zorun** [comment]
Draft PR: #814 

Still need to add a few testcases for some import corner cases:

- import bills without currency in a project with a default currency: should add the default currency to these bills
- import bills with a currency (the same for all bills) in a project with no default currency: should strip the currency on all bills
- import bills with several different currencies in a project with no default currency: should raise an error

Unfortunately, there are more corner cases: an imported file may contain some bills with currency and some bills without. We currently accept to import such a file, but it's not clear if other parts of I Hate Money can deal with it once imported.

**Comment by Glandos** [comment]
[**@zorun**](https://github.com/zorun) commented on [19 juil. 2021 à 00:13 UTC+2](https://github.com/spiral-project/ihatemoney/issues/806#issuecomment-882123284 "2021-07-18T22:13:22Z - Replied by Github Reply Comments"):
> Unfortunately, there are more corner cases: an imported file may contain some bills with currency and some bills without. We currently accept to import such a file, but it's not clear if other parts of I Hate Money can deal with it once imported.

In this case, bills without currency should have the default currency of the project to be consistent with the first point. If the project is set to "No Currency", then it will ultimately fail to import, since bills with different currencies would have been found.

**Comment by Mageek627** [comment]
Hello, I'm not sure where it's best to ask this, so let's try here:

This new currency feature (to which I'm really looking forward), is it strictly for known "fiat" currencies ? Would custom currencies be supported ? (For example: I owe you 2 HUGS) Will cryptocurrencies be supported ?

**Comment by zorun** [comment]
I have merged #814, thanks @Glandos for the review!
